### PR TITLE
fix(infra): fix the remaining terraform constant changes

### PR DESCRIPTION
**Pull request by bassrock**
## Goal

Cleanup some remaining infrastructure things.

* Move away from pocket-ci to a real Github App Bot (allows us to filter out Bot comments/etc) (all future comments will be by it)
     * Screenshot 2024-03-22 at 9 33 03 AM

* Fix shared lists event rule
* Fix dynamodb changing the global secondary index (https://github.com/hashicorp/terraform-provider-aws/issues/671)

### CODE REVIEW

**Comment by bassrock** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/Pocket/pocket-monorepo/331?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#331** <a href="https://app.graphite.dev/github/pr/Pocket/pocket-monorepo/331?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#329** <a href="https://app.graphite.dev/github/pr/Pocket/pocket-monorepo/329?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bassrock and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .circleci/common.yml [hunk 1]
```diff
@@ -180,6 +178,60 @@ commands:
             rc="${rc}}"
             echo "$rc" > ~/.terraformrc
 
+  setup_github_bot:
+    steps:
+      - run:
+          name: Get Github Bot Token
+          command: |
+            app_id=$GITHUB_APP_ID
```
Comment (bassrock) [inline .circleci/common.yml:186]: Sadly I couldn't find an orb to do this for us. But this script will take the Github Bot secrets and convert it to a regular access token like we already use, valid for 10 min.

i wish CircleCI let us ask for a github access token like Github Actions do, then this wouldnt be necessary 🤷‍♂️

File: infrastructure/annotations-api/src/dynamodb.ts [hunk 2]
```diff
@@ -49,6 +49,16 @@ export class DynamoDB extends Construct {
           },
         ],
       },
+      lifecycle: {
+        ignoreChanges: [
+          // Bug in terraform with DynamoDB and global secondary indexes
```
Comment (bassrock) [inline infrastructure/annotations-api/src/dynamodb.ts:54]: 😮‍💨

File: .circleci/common.yml (hunk 1 above)
Comment (kschelonka) [reply .circleci/common.yml:186]: Cool, we can make this an orb of our own later?

File: infrastructure/annotations-api/src/dynamodb.ts (hunk 2 above)
Comment (kschelonka) [reply infrastructure/annotations-api/src/dynamodb.ts:54]: from 2017? yikes

Comment (kschelonka) [reply infrastructure/annotations-api/src/dynamodb.ts:54]: Can you add this link to the PR description so that it gets tagged?

**Review by kschelonka — APPROVED** [review]

**Pull request merged**